### PR TITLE
Add vuid 06762 06763 06764 06765, Validate resolve image usage and format features

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Could not find a format to test 06763 and 06765

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -3683,6 +3683,16 @@ bool CoreChecks::ValidateCmdResolveImage(VkCommandBuffer commandBuffer, VkImage
         skip |= ValidateProtectedImage(cb_node.get(), dst_image_state.get(), func_name, vuid);
         vuid = is_2 ? "VUID-vkCmdResolveImage2-commandBuffer-01839" : "VUID-vkCmdResolveImage-commandBuffer-01839";
         skip |= ValidateUnprotectedImage(cb_node.get(), dst_image_state.get(), func_name, vuid);
+        vuid = is_2 ? "VUID-VkResolveImageInfo2-srcImage-06762" : "VUID-vkCmdResolveImage-srcImage-06762";
+        skip |= ValidateImageUsageFlags(src_image_state.get(), VK_IMAGE_USAGE_TRANSFER_SRC_BIT, true, vuid, func_name,
+                                        "VK_IMAGE_USAGE_TRANSFER_SRC_BIT");
+        vuid = is_2 ? "VUID-VkResolveImageInfo2-srcImage-06763" : "VUID-vkCmdResolveImage-srcImage-06763";
+        skip |= ValidateImageFormatFeatureFlags(dst_image_state.get(), VK_FORMAT_FEATURE_TRANSFER_SRC_BIT, func_name, vuid);
```
Comment (sjfricke) [inline layers/buffer_validation.cpp:3690]:
```suggestion
        skip |= ValidateImageFormatFeatureFlags(src_image_state.get(), VK_FORMAT_FEATURE_TRANSFER_SRC_BIT, func_name, vuid);
```

copy and paste error?

File: tests/vklayertests_command.cpp [hunk 2]
```diff
@@ -11160,3 +11163,76 @@ TEST_F(VkLayerTest, TestCommandBufferInheritanceWithInvalidDepthFormat) {
 
     m_errorMonitor->VerifyFound();
 }
+
+TEST_F(VkLayerTest, ResolveInvalidUsage) {
```
Comment (sjfricke) [inline tests/vklayertests_command.cpp:11167]: > Could not find a format to test 06763 and 06765

https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/docs/creating_tests.md#device-profile-format-feature

Comment (ziga-lunarg) [reply tests/vklayertests_command.cpp:11167]: I've just used this, but forgot about it, thanks!

**Review by Tony-LunarG — APPROVED** [review]
LGTM!

**Pull request merged**